dw-ripple.js

```diff
@@ -42,29 +68,80 @@ export class DwRipple extends Ripple {
     return {
 
       /**
-       * Set to `true` when ripple is to be shown in primary color.
-       */
-      primary: { type: Boolean, reflect: true },
-
-      /**
-       * Set to `true` when ripple is to be shown in secondary color
+       * Set to `true` if ripple should not be shown
        */
-      secondary: { type: Boolean, reflect: true },
+      disabled: { type: Boolean, reflect: true },
 
       /**
-       * Set to `true` if ripple should not be shown
+       * Set to true when ripple shows as an unbounded.
        */
-      disabled: { type: Boolean, reflect: true }
+      unbounded: { type: Boolean, reflect: true }
     }
   }
 
   constructor(){
     super();
-    this.primary=false;
-    this.secondary = false;
     this.disabled = false;
+    this.unbounded = false;
+    this.__activeRipple = this.__activeRipple.bind(this);
+  }
+
+
+  connectedCallback() {
+    super.connectedCallback && super.connectedCallback();
+    if(this.parentNode) {
+      this.parentNode.addEventListener('click', this.__activeRipple);
+    }
   }
 
+  disconnectedCallback() {
+    if(this.parentNode) {
+      this.parentNode.removeEventListener('click', this.__activeRipple);
+    }
+    super.disconnectedCallback && super.disconnectedCallback();
+  }
+
+  render() {
+    return html`
+    `;
+  }
+
+
+  /**
+   * Active ripple on button click event.
+   * @private
+   */
+  __activeRipple(event) {
+    let button = event.currentTarget;
+
+    //If parent is not found;
+    if(!button) {
```

Comment:
@shakil-mansuri-dream When does this happens? Why do we call the parent element as `button` here? `parentEl` is the proper name here.